lib/trivial.nix

```diff
@@ -332,4 +332,34 @@ rec {
   */
   isFunction = f: builtins.isFunction f ||
     (f ? __functor && isFunction (f.__functor f));
+
+  /* A trivial derivation that cannot be built. */
+  fakeDerivation = derivation {
+    name = "fake-derivation";
+    system = "fake-system";
+    builder = "fake-builder";
+  };
+
+  /* A fake store path.
+
+     This uses fakeDerivation, so the path can never be built; make sure
+     you don't depend on it. If it's gumming things up, you might need
+     to use builtins.unsafeDiscardStringContext.
+  */
+  fakeStorePath = "${fakeDerivation}";
+
+  /* A hash in the format used by store paths.
+
+     Don't rely on the value of this for anything; it's just used to
+     define storeHashLength and exported in the benevolent hope that you
+     might somehow find it useful.
+  */
+  fakeStoreHash = builtins.head
+    (builtins.split "-" (baseNameOf fakeStorePath));
+
+  /* The length of hashes in store paths.
+
+     Currently 32, but you never know!
+  */
+  storeHashLength = builtins.stringLength fakeStoreHash;
```

Comment:
The definitions for `fakeDerivation`, `fakeStorePath` and `fakeStoreHash` are only used by this definition here, and are arguably not of much use for other parts of the code base. So I suggest to just defined them as a `let in` under `storeHashLength`.

Reply:
Okay, made them local for now.

@GrahamcOfBorg test caddy-etag h2o-etag